integration/helpers/base_test.py

```diff
@@ -567,6 +567,13 @@ def verify_options_request(self, url, expected_status_code, headers=None):
             )
         return response
 
+    @retry(
+        stop=stop_after_attempt(5),
+        wait=wait_exponential(multiplier=1, min=16, max=64) + wait_random(0, 1),
+        retry=retry_if_exception_type(StatusCodeError),
+        after=after_log(LOG, logging.WARNING),
+        reraise=True,
+    )
```

Comment:
`verify_post_request` seems to be a method that gets called in multiple places. Is it safe to retry POST requests for all those cases? Would be safer if the caller can decide if retry or not.

I see we have similar retry logic above but those are Read-Only methods (e.g. GET).

Reply:
Thanks @samson-keung. In general it is not safe to retry POST requests. However this method I could see is only being used by [test_state_machine_with_api_single_quotes_input](https://github.com/aws/serverless-application-model/blob/7a0558f504912a37cda78dcf22e1ef98c8e1703f/integration/single/test_basic_api.py#L132) test and can be retried for this testcase. I don't see anyother places using this method.

Reply:
Thanks for checking!